assets/js/editor-components/product-control/index.tsx

```diff
@@ -142,29 +165,42 @@ const ProductControl = ( {
 		} else if ( showVariations ) {
 			return renderItemWithVariations;
 		}
-		return null;
+		return () => {
+			return null;
+		};
 	};
 
 	if ( error ) {
 		return <ErrorMessage error={ error } />;
 	}
 
 	const currentVariations =
-		variations && variations[ expandedProduct ]
+		variations && expandedProduct !== null && variations[ expandedProduct ]
 			? variations[ expandedProduct ]
 			: [];
 	const currentList = [ ...products, ...currentVariations ];
 
 	return (
 		<SearchListControl
 			className="woocommerce-products"
+			// @ts-expect-error - It looks like ProductResponse does not fulfill the SearchListItemProps interface. This needs to be fixed.
 			list={ currentList }
 			isCompact={ isCompact }
 			isLoading={ isLoading }
 			isSingle
-			selected={ currentList.filter( ( { id } ) =>
-				selected.includes( id )
-			) }
+			// @ts-expect-error - It looks like ProductResponse does not fulfill the SearchListItemProps interface. This needs to be fixed.
+			selected={ currentList.filter( ( { id } ) => {
```

Comment:
Based on types it's really hard to know what to do here. selected can be string | number[] | number according to the  type.

if its `number` then includes is not callable, if its array or string then it is and thats ok.

I can't find any tests for this component, so we might need to manual test here and look at adding some tests when I better understand the behaviour of this component.